Use symfony controllers for the endpoint

Chemaclass opened the pull request:
## 🔖 Changes

- Install Symfony to use the basic controllers logic
- Create a mapping between the user and its configuration

>**Note**: This PR would be better than this other one https://github.com/php-lightning/lnaddress/pull/17 

### 🎥  Demo

https://user-images.githubusercontent.com/5256287/230677048-75f6e564-cde2-4c01-9044-1fb9b975307b.mov

Chemaclass committed:
Create InvoiceController using Symfony

Chemaclass committed:
Remove Lightning class in favor of InvoiceFacade

Chemaclass committed:
Map backends to user->config

Chemaclass committed:
Install and use Symfony HttpClient

Chemaclass committed:
Validate user exists before using any endpoint api

JesusValeraDev commented on composer.json:
```diff
@@ -13,6 +17,21 @@
     },
+    "config": {
+        "platform": {
+            "php": "8.0.27"
```
You defined in `composer.json` as min PHP version the 8.0.2, you cannot define here a higher version 🤔

Chemaclass committed:
Move gacelaConfigFn to Kernel

Bashy commented:
This approach requires
* a shell access 
*  an unnecessary (in my opinion) additional backend that will be permanently listening (more resources, CPU, RAM, TCP port to listen on)
* more manipulation from the user before to be ready-to-use (more configuration to change is less easy to use)

in order to be used versus the other PR that could allow to be used on a simple web hosting plan and where the code is just executed when solicited.

It's a bit like a nuclear plant to light a bulb.

In my opinion user experience and resources optimization is slightly better with the previously closed PR #17 .

Chemaclass commented:
Oh! Actually, I am flattered somehow by this, because usually people like the idea of using symfony. So, you are the first person that likes my framework/Library(Gacela) more than the standard most popular framework symfony :)


However, there is a trick to this point: I didn't removed Gacela, and I just introduced some little dependencies from Symfony to use the controller, which is very easy and powerful.


To your three points (written in the PR):
1) there is no need to a shell access. It's based on the other approach.
2) there is no need different BE, it would be an identical solution as the previous one.
3) Similar DX, simpler in my point of view, because you don't need to override an entire file (copy-pasting another file) but just add a new backend in the global configuration in your app.


I reviewed again the last changes, and I found there are a few things that I could try to remove to make it even simpler. But in theory, whatever we used inside (with Symfony or without), I thought this is the best DX we wanted: a global config mapping instead of copy-pasting entire files.


Let me know if this is the case or you disagree for a particular reason that I might have missed.


I will focus on reduce the number of dependencies and the code I added. Worst scenario, we can rollback to what we had yesterday and think about another approach if you are still not convinced after trying it.
